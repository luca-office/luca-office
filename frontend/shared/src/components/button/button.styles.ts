import {css} from "@emotion/react"
import {ButtonVariant} from "../../enums"
import {
  boxHeightSmaller,
  buttonBackgroundDisabled,
  buttonBackgroundPrimary,
  buttonHeight,
  fontColorBright,
  fontFamily,
  primaryColor,
  spacingSmall,
  TextSize
} from "../../styles"

export const buttonStyle = (variant?: ButtonVariant) => ({
  button: css({
    minHeight: buttonHeight,
    width: variant === ButtonVariant.IconOnly ? buttonHeight : 165,
    borderRadius: 4,
    outline: "none",
    userSelect: "none",
    border:
      variant === ButtonVariant.Primary || variant === ButtonVariant.Ghost ? "none" : `1px solid  ${primaryColor}`,
    color: variant === ButtonVariant.Primary ? "white" : primaryColor,
    fontSize: TextSize.Medium,
    fontFamily: fontFamily,
    fontWeight: 300,
    display: "flex",
    alignItems: "center",
    justifyContent: "space-evenly",
    background:
      variant === ButtonVariant.Primary || variant === ButtonVariant.IconOnly ? buttonBackgroundPrimary : "none",
    cursor: "pointer",

    ":hover": {
      opacity: 0.8
    },

    ":disabled": {
      background: buttonBackgroundDisabled,
      cursor: "not-allowed",
      border: "none"
    }
  }),
  buttonContent: css({
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  }),
  icon: css({
    marginRight: variant === ButtonVariant.IconOnly ? 0 : spacingSmall
  }),
  loadingIndicator: css({
    height: boxHeightSmaller,
    width: boxHeightSmaller,
    color: fontColorBright,
    marginRight: spacingSmall
  })
})
