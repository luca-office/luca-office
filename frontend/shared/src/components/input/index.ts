export * from "./auto-complete-input"
export * from "./select-input"
export * from "./text-input"
