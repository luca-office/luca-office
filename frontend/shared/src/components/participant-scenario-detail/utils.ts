import {FreetextAnswerFragment} from "../../graphql/generated/FreetextAnswerFragment"
import {QuestionType} from "../../graphql/generated/globalTypes"
import {Questionnaire, RuntimeSurveyResults} from "../../models"
import {ParticipantQuestionAnswer, QuestionnaireParticipantResults} from "../questionnaire"

export const runtimeSurveyResultsToQuestionnaireParticipantResults = (
  invitationId: UUID,
  questionnaire: Questionnaire,
  runtimeSurveyResults: RuntimeSurveyResults,
  freeTextAnswers: FreetextAnswerFragment[]
): QuestionnaireParticipantResults => {
  const answers = runtimeSurveyResults.questionResults.reduce<Record<UUID, ParticipantQuestionAnswer>>(
    (answersAccumulator, result) => {
      const question = questionnaire.questions.find(question => question.id === result.questionId)
      const participantResult = result.participantResults.find(results => results.invitationId === invitationId)
      const freeTextAnswer = freeTextAnswers.find(
        answer => answer.questionId === result.questionId && answer.surveyInvitationId === invitationId
      )?.text

      switch (question?.questionType) {
        case QuestionType.SingleChoice: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.SingleChoice,
              questionId: question.id,
              selectedAnswerId: participantResult?.selectedAnswerIds[0] ?? "",
              isFreetextAnswerSelected: participantResult?.wasFreetextAnswerSelected ?? false,
              freeTextAnswer: freeTextAnswer
            }
          }
        }

        case QuestionType.MultipleChoice: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.MultipleChoice,
              questionId: question.id,
              selectedAnswerIds: participantResult?.selectedAnswerIds ?? [],
              isFreetextAnswerSelected: participantResult?.wasFreetextAnswerSelected ?? false,
              freeTextAnswer: freeTextAnswer
            }
          }
        }

        case QuestionType.FreeText: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.FreeText,
              questionId: question.id,
              freeTextAnswer: freeTextAnswer ?? ""
            }
          }
        }

        default:
          return answersAccumulator
      }
    },
    {}
  )

  return {questionnaireId: runtimeSurveyResults.questionnaireId, answers}
}
