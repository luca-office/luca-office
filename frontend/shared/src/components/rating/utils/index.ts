export * from "./coding-criterion"
export * from "./coding-dimension"
export * from "./coding-item"
export * from "./question"
export * from "./score"
export * from "./scoring-type"
