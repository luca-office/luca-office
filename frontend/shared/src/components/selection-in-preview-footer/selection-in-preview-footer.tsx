import {css} from "@emotion/react"
import {CSSInterpolation} from "@emotion/serialize"
import * as React from "react"
import {ButtonVariant, HeadingLevel, IconName} from "../../enums"
import {
  backgroundColorLight,
  borderRadius,
  Flex,
  insetShadow,
  spacingMedium,
  spacingSmall,
  spacingTiny
} from "../../styles"
import {LucaI18nLangKey, useLucaTranslation} from "../../translations"
import {Button, Heading, Icon, Label, Text} from "../"

export interface SelectionInPreviewFooterProps {
  readonly customContent?: JSX.Element
  readonly customLabelStyles?: CSSInterpolation
  readonly headingKey: LucaI18nLangKey
  readonly icon?: IconName
  readonly onConfirm?: () => void
  readonly textKey?: LucaI18nLangKey
  readonly title?: string
}

export const SelectionInPreviewFooter: React.FC<SelectionInPreviewFooterProps> = ({
  customContent,
  customLabelStyles,
  headingKey,
  icon,
  onConfirm,
  textKey,
  title
}) => {
  const {t} = useLucaTranslation()
  return (
    <div css={styles.footer}>
      <Label customStyles={customLabelStyles} label={t(headingKey)} />

      <div className="card-wrapper" css={styles.card}>
        <div css={Flex.row}>
          {icon && <Icon customStyles={styles.icon} name={icon} />}
          {title !== undefined && <Heading level={HeadingLevel.h3}>{title}</Heading>}
        </div>
        <div css={styles.buttonRow}>
          {textKey && <Text>{t(textKey)}</Text>}
          {customContent === undefined && (
            <Button onClick={onConfirm} variant={ButtonVariant.Primary}>
              {t("continue_button")}
            </Button>
          )}
        </div>
        {customContent}
      </div>
    </div>
  )
}

const styles = {
  footer: css({
    backgroundColor: backgroundColorLight
  }),
  buttonRow: css(Flex.row, {
    justifyContent: "space-between"
  }),
  icon: css({
    marginRight: spacingSmall
  }),
  card: css({
    margin: spacingSmall,
    boxShadow: insetShadow,
    padding: spacingMedium,
    backgroundColor: "white",
    borderRadius: borderRadius
  })
}
