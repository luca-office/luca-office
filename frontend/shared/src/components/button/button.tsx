import {CSSInterpolation} from "@emotion/serialize"
import * as React from "react"
import {ButtonHTMLAttributes, DetailedHTMLProps} from "react"
import {ButtonVariant, IconName} from "../../enums"
import {CustomStyle, iconBrightColor} from "../../styles"
import {Icon} from ".."
import {LoadingIndicator} from "../loading-indicator/loading-indicator"
import {buttonStyle} from "./button.styles"

interface Props extends DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
  readonly customIconStyles?: CSSInterpolation
  readonly icon?: IconName
  readonly iconColor?: string
  readonly stopEventPropagation?: boolean
  readonly variant?: ButtonVariant
  readonly isLoading?: boolean
}

export const Button: React.FC<Props & CustomStyle> = ({
  children,
  className,
  customIconStyles,
  customStyles,
  disabled,
  icon,
  isLoading = false,
  iconColor,
  onClick,
  stopEventPropagation = false,
  title,
  variant
}) => {
  const styles = buttonStyle(variant)

  const handleClick = (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    if (stopEventPropagation) {
      evt.stopPropagation()
    }

    if (onClick) {
      onClick(evt)
    }
  }

  return (
    <button
      disabled={disabled || isLoading}
      css={[styles.button, customStyles]}
      className={className}
      title={title}
      onClick={handleClick}>
      <div css={styles.buttonContent}>
        {isLoading && <LoadingIndicator customStyles={styles.loadingIndicator} />}
        {icon !== undefined && !isLoading ? (
          <Icon color={iconColor ?? iconBrightColor} customStyles={[styles.icon, customIconStyles]} name={icon} />
        ) : null}
        {variant !== ButtonVariant.IconOnly ? children : null}
      </div>
    </button>
  )
}

Button.defaultProps = {
  variant: ButtonVariant.Primary
}
