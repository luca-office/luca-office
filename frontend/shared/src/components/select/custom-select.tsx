import {CSSInterpolation} from "@emotion/serialize"
import * as React from "react"
import {
  ActionMeta,
  components,
  CSSObjectWithLabel,
  default as Select,
  MenuPosition,
  MultiValue,
  OnChangeValue,
  SelectComponentsConfig,
  SelectInstance,
  SingleValue,
  StylesConfig
} from "react-select"
import {useClipboard} from "use-clipboard-copy"
import {IconName, IconPosition} from "../../enums"
import {SelectOption} from "../../models"
import {CustomStyle, Flex, fontColor} from "../../styles"
import {LucaI18nLangKey, useLucaTranslation} from "../../translations"
import {isDefined} from "../../utils/common"
import {FormFieldLabel, Icon, Tooltip, TooltipConfig} from ".."
import {customSelectStyles as inputStyles} from "./custom-select.style"

export interface SelectOptionCustomized extends Omit<SelectOption, "selected"> {
  iconName?: IconName
  iconColor?: string
  iconPosition?: IconPosition
  renderOptionLabel?: (label: string) => React.ReactNode
  readonly showCopyToClipboard?: boolean
}

type SelectComponentRef =
  | ((instance: SelectInstance<SelectOptionCustomized> | null) => void)
  | React.RefObject<SelectInstance<SelectOptionCustomized>>
  | null
  | undefined

interface Props extends CustomStyle {
  readonly onChange: (value: string) => void
  readonly optionList: SelectOptionCustomized[]
  readonly value: string
  readonly disabled?: boolean
  readonly clearable?: boolean
  readonly labelKey?: LucaI18nLangKey
  readonly name?: string
  readonly placeholderKey?: LucaI18nLangKey
  readonly ref?: SelectComponentRef
  readonly searchable?: boolean
  readonly showCopyToClipboard?: boolean
  readonly showIconOnHover?: boolean
  readonly onCopyToClipboard?: (value: string) => void
  readonly customCopyToClipboardIconStyle?: CSSInterpolation
  readonly tooltipConfig?: TooltipConfig
  readonly menuPosition?: MenuPosition
  readonly zIndexOfExpandedMenu?: number
  readonly menuPortalTarget?: HTMLElement | null
  readonly hideDropdownIndicatorWhenDisabled?: boolean
  readonly customInnerContainerStyle?: React.CSSProperties
  readonly customSingleValueStyle?: React.CSSProperties
  readonly customOptionStyle?: React.CSSProperties
  readonly customIconColor?: string
}

export const CustomSelect: React.FunctionComponent<Props> = React.forwardRef(
  (
    {
      value,
      onChange,
      labelKey,
      optionList,
      customStyles,
      name,
      placeholderKey,
      searchable,
      clearable,
      onCopyToClipboard,
      tooltipConfig,
      disabled = false,
      showCopyToClipboard = false,
      showIconOnHover = false,
      customCopyToClipboardIconStyle,
      menuPosition = "fixed",
      zIndexOfExpandedMenu,
      menuPortalTarget,
      hideDropdownIndicatorWhenDisabled = false,
      customInnerContainerStyle,
      customSingleValueStyle,
      customOptionStyle,
      customIconColor
    },
    ref
  ) => {
    const {t} = useLucaTranslation()

    const [isHovered, setIsHovered] = React.useState(false)

    const {copy: copyToClipboard} = useClipboard()

    const hasLabel = isDefined(labelKey)

    // const handleChange = (value: OnChangeValue<SelectOptionCustomized, boolean>, actionMeta: ActionMeta<unknown>) => {
    //   if ((value as SelectOptionCustomized)?.value) {
    //     onChange((value as SelectOptionCustomized).value)
    //   }
    // }

    const handleChange = (
      value: SingleValue<SelectOptionCustomized> | MultiValue<SelectOptionCustomized>,
      actionMeta: ActionMeta<SelectOptionCustomized>
    ) => {
      if ((value as SingleValue<SelectOptionCustomized>)?.value) {
        onChange((value as SelectOptionCustomized).value)
      }
    }
    const selectedValue: SelectOptionCustomized | undefined = optionList.find(option => option.value === value)
    const showDuplicateIcon = disabled && (!showIconOnHover || isHovered) && showCopyToClipboard
    const hideDropdownIndicator = disabled && hideDropdownIndicatorWhenDisabled

    const copySelectionToClipboard = () => {
      copyToClipboard(selectedValue?.label)
      onCopyToClipboard?.(selectedValue?.label ?? "")
    }

    const styleConfig: StylesConfig<SelectOptionCustomized, any, any> = {
      control: (styles: CSSObjectWithLabel) => ({
        ...styles,
        ...(inputStyles.control(hasLabel, showDuplicateIcon) as CSSObjectWithLabel),
        ...customInnerContainerStyle
      }),
      indicatorSeparator: (styles: CSSObjectWithLabel) => ({
        ...styles,
        ...(!clearable && !searchable ? {display: "none"} : {})
      }),
      option: (styles: CSSObjectWithLabel, state) => ({
        ...styles,
        ...inputStyles.option,
        ...(state.isSelected ? inputStyles.optionSelected : {}),
        ...(state.isFocused ? inputStyles.optionFocused : {}),
        ...customOptionStyle
      }),
      singleValue: (styles: CSSObjectWithLabel) => ({...styles, ...inputStyles.option, ...customSingleValueStyle})
    }

    // TODO LUCA-2757 remove any
    const componentsConfig: SelectComponentsConfig<SelectOptionCustomized, any, any> = {
      DropdownIndicator: props =>
        !showDuplicateIcon && !hideDropdownIndicator ? (
          <components.DropdownIndicator {...props}>
            <Icon name={IconName.TriangleDown} color={customIconColor !== undefined ? customIconColor : fontColor} />
          </components.DropdownIndicator>
        ) : null,
      Option: props => (
        <components.Option {...props}>
          {props.data.iconName && props.data.iconPosition !== IconPosition.RightAlign && (
            <Icon name={props.data.iconName} color={props.data.iconColor} customStyles={inputStyles.optionIcon} />
          )}
          {props.data.renderOptionLabel ? props.data.renderOptionLabel(props.label) : props.label}
          {props.data.iconName && props.data.iconPosition === IconPosition.RightAlign && (
            <Icon name={props.data.iconName} color={props.data.iconColor} customStyles={inputStyles.optionIconRight} />
          )}
        </components.Option>
      ),
      SingleValue: props => (
        <components.SingleValue className={"single-value"} {...props}>
          {props.data.iconName && (
            <Icon name={props.data.iconName} color={props.data.iconColor} customStyles={inputStyles.optionIcon} />
          )}
          {props.data.renderOptionLabel ? props.data.renderOptionLabel(props.data.label) : props.data.label}
        </components.SingleValue>
      )
    }

    return (
      <div data-testid="custom-select" css={[inputStyles.container(hasLabel), customStyles]}>
        <div css={hasLabel && !!tooltipConfig ? Flex.row : undefined}>
          {hasLabel && <FormFieldLabel textKey={labelKey as LucaI18nLangKey} />}
          {!!tooltipConfig && (
            <Tooltip
              inactive={!tooltipConfig.labelKey}
              title={tooltipConfig.labelKey ? t(tooltipConfig.labelKey) : ""}
              text={tooltipConfig.textKey ? t(tooltipConfig.textKey) : undefined}
              icon={tooltipConfig.iconName}>
              <Icon customStyles={inputStyles.infoIcon} name={IconName.Information} />
            </Tooltip>
          )}
        </div>

        <div
          css={inputStyles.content(zIndexOfExpandedMenu)}
          onMouseEnter={() => setIsHovered(true)}
          onMouseLeave={() => setIsHovered(false)}>
          <Select
            ref={ref}
            className={"custom-select"}
            onChange={handleChange}
            options={optionList}
            value={selectedValue}
            isDisabled={disabled}
            menuPlacement="auto"
            menuPosition={menuPosition}
            name={name}
            placeholder={t(placeholderKey || "select__placeholder")}
            styles={styleConfig}
            components={componentsConfig}
            isSearchable={!!searchable}
            isClearable={!!clearable}
            noOptionsMessage={() => t("select__empty")}
            menuPortalTarget={menuPortalTarget}
          />
          {showDuplicateIcon && (
            <Tooltip customStyles={inputStyles.duplicateIconWrapper} title={t("clipboard__copy")}>
              <Icon
                customStyles={[inputStyles.duplicateIcon, customCopyToClipboardIconStyle]}
                name={IconName.Duplicate}
                onClick={copySelectionToClipboard}
              />
            </Tooltip>
          )}
        </div>
      </div>
    )
  }
)
