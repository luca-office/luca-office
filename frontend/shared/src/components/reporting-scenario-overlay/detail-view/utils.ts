import {TableEntity} from "../../../components/rating/models"
import {CodingDimension, CodingItem, CodingItemResultByItemId} from "../../../models"

export const itemToTableEntity = (
  item: CodingItem,
  codingItemResultsByItem: CodingItemResultByItemId
): TableEntity => ({
  id: item.id,
  title: item.title,
  position: item.position,
  score: codingItemResultsByItem[item.id].score,
  maxScore: codingItemResultsByItem[item.id].maximumScore,
  averageScore: codingItemResultsByItem[item.id].averageScore,
  rated: true
})

export const dimensionToTableEntity = (
  dimension: CodingDimension,
  items: CodingItem[],
  codingItemResultsByItem: CodingItemResultByItemId
): TableEntity => ({
  id: dimension.id,
  title: dimension.title,
  position: dimension.position,
  score: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].score, 0),
  maxScore: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].maximumScore, 0),
  averageScore: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].averageScore, 0),
  rated: true
})
