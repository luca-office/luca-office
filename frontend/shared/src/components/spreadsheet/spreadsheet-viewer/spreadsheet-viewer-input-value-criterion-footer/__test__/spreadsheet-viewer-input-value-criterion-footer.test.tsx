import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {mockedFormMethods} from "../../../../../../tests/utils/form-methods-mock"
import {Option} from "../../../../../utils"
import {Button} from "../../../../button/button"
import {TextInput} from "../../../../input"
import {CustomSelect} from "../../../../select/custom-select"
import {SelectionInPreviewFooter} from "../../../../selection-in-preview-footer/selection-in-preview-footer"
import {
  InputValueCriterionSpreadsheetFooter,
  SpreadsheetInputValueFooterProps,
  SpreadsheetInputValueSelectionType
} from "../spreadsheet-viewer-input-value-criterion-footer"

const defaultProps: SpreadsheetInputValueFooterProps = {
  fileName: "fileName.xls",
  inputValueAutomatedCriterionConfig: {
    formMethods: mockedFormMethods,
    onConfirm: jest.fn(),
    selectedCellName: Option.none()
  },
  selectedSelectionType: SpreadsheetInputValueSelectionType.SpecificCell,
  setSelectedSelectionType: jest.fn()
}

const getComponent = (props?: Partial<SpreadsheetInputValueFooterProps>) => (
  <InputValueCriterionSpreadsheetFooter {...{...defaultProps, ...props}} />
)

describe("spreadsheet-input-value-footer", () => {
  it("renders correctly", () => {
    const component = create(getComponent())
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure - specific cell", () => {
    const component = mount(getComponent())
    expect(component.find(SelectionInPreviewFooter)).toHaveLength(1)
    expect(component.find(TextInput)).toHaveLength(2)
    expect(component.find(CustomSelect)).toHaveLength(1)
    expect(component.find(Button)).toHaveLength(1)
  })
  it("has correct structure - whole table", () => {
    const component = mount(getComponent({selectedSelectionType: SpreadsheetInputValueSelectionType.WholeTable}))
    expect(component.find(SelectionInPreviewFooter)).toHaveLength(1)
    expect(component.find(TextInput)).toHaveLength(1)
    expect(component.find(CustomSelect)).toHaveLength(1)
    expect(component.find(Button)).toHaveLength(1)
  })
  it("shows correct cell - no cell selected", () => {
    const component = mount(
      getComponent({
        inputValueAutomatedCriterionConfig: {
          ...defaultProps.inputValueAutomatedCriterionConfig,
          selectedCellName: Option.none()
        }
      })
    )
    expect(component.find(TextInput).last().prop("value")).toEqual("A1")
  })
  it("shows correct cell -  cell selected", () => {
    const component = mount(
      getComponent({
        inputValueAutomatedCriterionConfig: {
          ...defaultProps.inputValueAutomatedCriterionConfig,
          selectedCellName: Option.of("C2")
        }
      })
    )
    expect(component.find(TextInput).last().prop("value")).toEqual("C2")
  })
})
