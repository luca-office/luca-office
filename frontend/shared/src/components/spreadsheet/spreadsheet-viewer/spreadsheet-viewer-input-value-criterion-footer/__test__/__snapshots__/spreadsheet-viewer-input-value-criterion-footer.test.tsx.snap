// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spreadsheet-input-value-footer renders correctly 1`] = `
<div
  className="css-qs6aqy-SelectionInPreviewFooter"
>
  <div
    className="css-1nc3kou-Label"
  >
    coding_models__automated_item_input_value_preview_spreadsheet_title
  </div>
  <div
    className="card-wrapper css-160r133-SelectionInPreviewFooter"
  >
    <div
      className="css-1gm10pm-SelectionInPreviewFooter"
    >
      <div
        className="css-1lrzij5-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1f6p2vj-Heading"
      >
        fileName.xls coding_models__automated_item_input_value_preview_spreadsheet_title_check_file_input
      </div>
    </div>
    <div
      className="css-100l0bw-SelectionInPreviewFooter"
    >
      <div
        className="css-fmthw-Text"
      >
        coding_models__automated_item_input_value_preview_spreadsheet_text_table
      </div>
    </div>
    <div
      className="css-dizc7g-InputValueCriterionSpreadsheetFooter"
    >
      <div
        className="css-17as3x0"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="css-13s2sfk-Heading"
        >
          coding_models__automated_item_input_value_value
        </div>
        <div
          className="css-1unfqpo"
        >
          <input
            className="css-pkbj54"
            defaultValue=""
            disabled={false}
            name="value"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="css-1wc1yin"
        data-testid="custom-select"
      >
        <div
          className="css-0"
        />
        <div
          className="css-18vooac"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="custom-select"
          >
            MockedSelect
          </div>
        </div>
      </div>
      <div
        className="css-1e6jpai-InputValueCriterionSpreadsheetFooter"
      >
        <div
          className="css-17as3x0"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="css-1unfqpo"
          >
            <input
              className="css-pkbj54"
              defaultValue="A1"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
            />
          </div>
        </div>
      </div>
      <button
        className="css-13t6j5m-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          confirm_button
        </div>
      </button>
    </div>
  </div>
</div>
`;
